examples/caa/caa.ipynb

```diff
@@ -0,0 +1,590 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Contrastive Activation Addition\n",
+    "\n",
+    "This notebook aims to reproduce the workflow defined in [Contrastive Activation Addition](https://arxiv.org/abs/2312.06681) for extracting steering vectors from input. The official codebase can be found [here](https://github.com/nrimsky/CAA)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To be consistent with CAA, we run on Llama-2 chat models of sizes 7b and 13b. These can be downloaded through Huggingface Transformers but require you to have first applied for access. \n",
+    "\n",
+    "- If you have downloaded the models to the standard location, you can skip the next cell.\n",
+    "- If you have downloaded the models to a custom path, you can set TRANSFORMERS_CACHE to point to your download path. \n",
+    "- If you have not downloaded the models, you will need to apply for access [here](https://ai.meta.com/resources/models-and-libraries/llama-downloads/). Once you have gotten access, you can set your HF_TOKEN to download the models. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# %env HF_TOKEN = <YOUR_HF_TOKEN>\n",
+    "# %env TRANSFORMERS_CACHE = <YOUR_CACHE_DIR>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/daniel/steering-vectors/.venv/lib/python3.10/site-packages/transformers/utils/hub.py:124: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os \n",
+    "from transformers import AutoModelForCausalLM, AutoTokenizer\n",
+    "token = os.getenv(\"HF_TOKEN\", None)\n",
+    "\n",
+    "def get_model_and_tokenizer(model_name: str):\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(model_name, use_auth_token=token)\n",
+    "    # Note: you must have installed 'accelerate', 'bitsandbytes' to load in 8bit\n",
```

Comment:
We should just `!pip install` this stuff at the top of the notebook, along with `steering-vectors`.